### PR TITLE
Minchingtonak/macro preload

### PR DESCRIPTION
**Pull request by minchingtonak**

### CODE REVIEW

File: src/org/hammerhead226/sharkmacro/Parser.java [hunk 1]
```diff
@@ -238,4 +277,12 @@ private static boolean touchDirectory(String directory) {
 		return true;
 	}
 
+	private static String format(String directory, String filename) {
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/Parser.java]: You probably meant to doc this.

Good call on a new method, though.

File: src/org/hammerhead226/sharkmacro/Parser.java [hunk 2]
```diff
@@ -102,13 +101,55 @@ protected boolean writeToFile(ArrayList<String[]> data) {
...
 		}
 
+		return cache();
+	}
+
+	/**
+	 * Add a file to the cache.
+	 * 
+	 * @param filename
+	 *            the file to cache
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/Parser.java]: Probably wanted a `@param` for `directory`.

File: src/org/hammerhead226/sharkmacro/Parser.java [hunk 3]
```diff
@@ -102,13 +101,55 @@ protected boolean writeToFile(ArrayList<String[]> data) {
...
+	 */
+	protected static void cache(String directory, String filename) {
+		if (cache.containsKey(filename)) {
+			DriverStation.getInstance();
+			DriverStation.reportWarning("Tried to cache an already cached file!", false);
+			return;
+		}
+
+		cache.put(filename, parseCSV(filename));
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/Parser.java]: I think this should also return the `List<String[]>`, to be consistent.

File: src/org/hammerhead226/sharkmacro/actions/ActionListParser.java [hunk 4]
```diff
@@ -7,6 +7,8 @@
 import org.hammerhead226.sharkmacro.Parser;
 import org.hammerhead226.sharkmacro.motionprofiles.Profile;
 
+import com.sun.xml.internal.bind.v2.runtime.reflect.opt.Const;
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/actions/ActionListParser.java]: Intended?

File: src/org/hammerhead226/sharkmacro/motionprofiles/ProfileParser.java [hunk 5]
```diff
@@ -66,11 +66,12 @@ public boolean writeToFile(Profile profile) {
...
 	public Profile toObject(TalonSRX leftTalon, TalonSRX rightTalon, int leftPidSlotIdx, int rightPidSlotIdx) {
 		List<String[]> profileRaw = readFromFile();
-		
+
 		if (profileRaw == null) {
 			DriverStation.getInstance();
 			DriverStation.reportError("Tried to load nonexistant Profile from name: " + super.filename, false);
-			return new Profile(new double[0][0], new double[0][0], leftTalon, rightTalon, leftPidSlotIdx, rightPidSlotIdx);
+			return new Profile(new double[0][0], new double[0][0], leftTalon, rightTalon, leftPidSlotIdx,
+					rightPidSlotIdx);
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/motionprofiles/ProfileParser.java]: ???

File: src/org/hammerhead226/sharkmacro/Parser.java (hunk 1 above)
Comment (minchingtonak) [reply src/org/hammerhead226/sharkmacro/Parser.java]: Oops, you right.

File: src/org/hammerhead226/sharkmacro/actions/ActionListParser.java (hunk 4 above)
Comment (minchingtonak) [reply src/org/hammerhead226/sharkmacro/actions/ActionListParser.java]: No sir.

File: src/org/hammerhead226/sharkmacro/motionprofiles/ProfileParser.java (hunk 5 above)
Comment (minchingtonak) [reply src/org/hammerhead226/sharkmacro/motionprofiles/ProfileParser.java]: Prolly Eclipse auto-formatting idek.

**Pull request merged**